### DISCUSSION TITLE
Additional features that may expedite the project,

### DISCUSSION THREAD

**Discussion by Alignment-Lab-AI** [webgrip/PuttyGPT#18] [General]
hello, i am currently in the process of training a llm, that will, when complete, be optimized specifically for the purpose of piloting one of these task oriented agii systems.

i wont disclose publicly what exactly is going in to training it but if you'd like to communicate privately you might be quite excited to  hear.

aside from that though, ive been working on constructing my own system like this but ive been so busy working on my model im a fair bit behind the wave. you're the only one who seems to be exclusively focusing on really encapsulating the scope of what is possible and i have a few ideas id like to hear your interest in implementing. im considering focusing the last bit of fine tuning for my model on giving it the knowledge to effectively use langchain and the tools at hand,

for example there are many examples of how to implement things like
local vector storage in a more robust way than chroma,  (https://milvus.io/ or https://github.com/ksjpswaroop/txtai)
arbitrary script execution (https://github.com/d3n7/GPT-4-Unlimited-Tools)
browse the internet in the CLI without needing to use SERPs   (https://packages.ubuntu.com/search?keywords=elinks, https://packages.ubuntu.com/search?keywords=lynx, https://packages.ubuntu.com/search?keywords=w3m)
 open sourced and very light weight, realistic voice models with custom editing of tonality (https://github.com/neonbjb/tortoise-tts)

even the main repository now has the ability t o use local models - in my own personal testing i was able to achieve a very high standard of results using 13b+7b vicuna on CPU by having one very focused on being only an operator of the tools, and the larger one being responsible to making the final choices. i was getting very fast inference speed on around 24 gigs of vram (with wsl ) and i imagine incorporating deepspeed, would make a large difference. i  dont have as much free time as id like with my primary focus being on developing and training models, but i do really want to provide a solution that is tuned for the specific task of piloting a system just like this one and with an explicit understanding of how to use a standardized set of tools to do it most effectively.

please reach out either here or via Autometa@alignmentlab.ai to discuss further :)

**Comment by Ryangr0** [comment]
@Alignment-Lab-AI I will send you a dm. Maybe we can work together!

**Comment by moover-shaker** [comment]
@Alignment-Lab-AI I am also interested in this and may be able to contribute a bit, at least in terms of testing and feedback. Will send separate email.
